src/ValueHandler/ImmutableSlugValueHandler.php

```diff
@@ -131,19 +131,21 @@ private function getDeduplicatedSlug(
         if ($increment > self::MAX_DEDUPLICATION_INCREMENT) {
             throw new \RuntimeException('Maximum slug deduplication increment reached.');
         }
+        $slugToCheck = $slug;
```

Comment:
I'd rename this variable $deduplicatedSlug

Reply:
Are you sure? We still don't know if this slug is duplicated. Wouldn't IncrementalSlug be better?

Reply:
Yes I am. At the time of the first assignment we don't know if it's deduplicated but then the variable is written again with a deduplicated slug and is returned. So that variable will become for sure the deduplicated slug.